packages/web-app-files/src/components/AppBar/AppBar.vue

```diff
@@ -496,7 +530,57 @@ export default {
 
       this.fileFolderCreationLoading = false
     },
+    async addAppProviderFile(fileName) {
+      try {
+        const path = pathUtil.join(this.currentPath, fileName)
+        const url = '/app/new?filename=' + path
+        const headers = new Headers()
+        headers.append('Authorization', 'Bearer ' + this.getToken)
+        headers.append('X-Requested-With', 'XMLHttpRequest')
+        const response = await fetch(encodeURI(url), {
+          method: 'POST',
+          headers
+        })
 
+        await response.json()
+
+        if (!response.ok) {
+          const message = `An error has occured: ${response.status}`
+          throw new Error(message)
+        }
+
+        let resource
+        if (this.isPersonalRoute) {
+          resource = await this.$client.files.fileInfo(path, DavProperties.Default)
+        } else {
+          resource = await this.$client.publicFiles.getFileInfo(
+            path,
+            this.publicLinkPassword,
+            DavProperties.Default
+          )
+        }
+        resource = buildResource(resource)
+        this.UPSERT_RESOURCE(resource)
+        this.$_fileActions_triggerDefaultAction(resource)
+        this.hideModal()
+        if (this.isPersonalRoute) {
+          this.loadIndicators({
+            client: this.$client,
+            currentFolder: this.currentFolder.path
+          })
+        }
+        setTimeout(() => {
+          this.setFileSelection([resource])
+          this.scrollToResource(resource)
+        })
+      } catch (error) {
+        this.showMessage({
+          title: this.$gettext('Creating file failed…'),
+          desc: error,
```

Comment:
I'd prefer to leave out the error, only use the title. Make sure that it gets printed to the console. Presenting http error codes to users is not good UX. 😁

Reply:
I removed it from the message here but the same error message is used in upstream addNewFiles()

Reply:
The one for `addNewFiles` is not good either 😁 Feel free to adapt in this PR as well, but also fine to do it in a separate one. Thanks for changing it!